packages/server-wallet/src/match.ts

```diff
@@ -0,0 +1,27 @@
+export {Either, left, right} from 'fp-ts/lib/Either';
+export {Option, none, some} from 'fp-ts/lib/Option';
+
+import {Protocol, ChannelState, ProtocolResult} from './protocols/state';
+
+type Partitioner<T, PS> = (ps: PS) => T;
+
+export const strictMatch = <PS extends ChannelState, T extends string>(
+  partitioner: Partitioner<T, PS>,
+  handlers: Record<T, Protocol<PS>>
+) => (ps: PS): ProtocolResult => handlers[partitioner(ps)](ps);
+
+export const match = <PS extends ChannelState, T extends string>(
+  partitioner: Partitioner<T, PS>,
+  handlers: (Partial<Record<T, Protocol<PS>>> & {Default: Protocol<PS>}) | Record<T, Protocol<PS>>
+) => (ps: PS): ProtocolResult => {
+  const handler = handlers[partitioner(ps)];
+  if (handler) {
+    return handler(ps);
```

Comment:
It'd be nice to allow the handlers to be sync or async functions. With this change, then `match` will still return a `Promise`
```suggestion
    return await handler(ps);
```

Reply:
With the change it looks like `match` is not returning a `Promise` which is what I would expect?

Reply:
https://github.com/statechannels/statechannels/commit/3801967b8cec16dfd2f87eb62c118c0ea6ad6e29

Maybe we should do something like that?